### ISSUE TITLE
Can language be set without including the JS translation file?

### ISSUE THREAD

**Issue by ajssd** [bgrins/spectrum#298]
Is there a way to set the language of the Spectrum color picker without having to directly include the JS translation file, e.g. i18n/jquery.spectrum-de.js ?

Ideally, this would just work using the standard i18n requirejs plugin, i.e. https://github.com/requirejs/i18n.   Then the Spectrum code could just 'require' something like 'i18n!spectrumStrings' and it would automatically pick up the language centrally configured.

**Comment by bgrins** [comment]
Right now you have to load the extra script.  I wouldn't be opposed to also supporting a standard require JS loading technique but I don't want to switch entirely since:

1) I don't want to rely on people having requirejs included for people who want a localized colorpicker.
2) Some people like having the separate script since it works similar to some other js plugins.  How it works now is modeled after the jquery UI datepicker: http://jqueryui.com/datepicker/#localization.

That said, if we could modify the plugin to also support this pattern it sounds like it could help.  Do you have an idea would need to be done to get to a point where requiring `i18n!spectrumStrings` would work?

One difficulty looking at http://requirejs.org/docs/api.html#i18n is that we may have to change the directory structure from `i18n/spectrum-de.js` to `i18n/de/spectrumStrings.js`, which would break backwards-compat.  Not sure if there is a config option that allows us to have a different file format.

**Comment by ajssd** [comment]
Yes, I think you are right -- the file format is different for requirejs/i18n, as is the directory requirement.  With requirejs/i18n, you put your file in a 'de' directory like you said, but the contents of the file are pretty simple -- 

define({
	"CHOOSE": "Choose",
	"CANCEL": "Cancel"
})

I am not sure which way is more standard, using the jQuery approach or using requirejs/i18n.  Perhaps someone with more experience can chime in.

**Comment by bgrins** [comment]
> I am not sure which way is more standard, using the jQuery approach or using requirejs/i18n. Perhaps someone with more experience can chime in.

I'm also interested in this, I'd like to do it in a way that causes the least amount of friction for the majority of people (even if it causes backwards-compat issues)
